Yokogawa GS200 Instrument

dennisfeng2 opened the pull request:
Added the beginning of the new instrument: Yokogawa GS200

dennisfeng2 commented:
I've noticed that there are some failed tests/checks with this pull request, but after taking a look at the error messages, it's still unclear to me whether these issues are coming from bugs in the pull request itself. For example, some of the errors involve import errors of VXI11Adapter and HP34410A, which my code doesn't seem to touch at all. Could someone point out to me why these errors are occurring, or anything I can fix from my end, so we can make progress on this PR?

cjermain commented:
Please merge the latest master branch into yours to fix the Travis CI build failure.

dennisfeng2 commented:
Okay, merged with latest master and added the RST documentation. Could you take a look?

dennisfeng2 commented on pymeasure/instruments/yokogawa/yokogawags200.py:
```diff
@@ -0,0 +1,117 @@
+    """ Represents the Yokogawa GS200 source and provides a high-level interface for interacting with the instrument.
+    """
+
+    source_enabled = Instrument.control("OUTPut:STATe?", "OUTPut:STATe %d",
```
done

dennisfeng2 commented on pymeasure/instruments/yokogawa/yokogawags200.py:
```diff
@@ -0,0 +1,117 @@
+    source_enabled = Instrument.control("OUTPut:STATe?", "OUTPut:STATe %d",
+                                        """A boolean property that controls whether the source is enabled, takes """
+                                        """values True or False. """,
+                                        validator=strict_discrete_set, values={True: 1, False: 0}, map_values=True)
```
done

dennisfeng2 commented on pymeasure/instruments/yokogawa/yokogawags200.py:
```diff
@@ -0,0 +1,117 @@
+    """
+
+    source_enabled = Instrument.control("OUTPut:STATe?", "OUTPut:STATe %d",
+                                        """A boolean property that controls whether the source is enabled, takes """
```
done

dennisfeng2 commented:
Hi, this is another old PR, and I also finally got around to addressing the comments, and I'd like to push this forward. Are there any more code changes you'd like me to make, before we can merge it?

dennisfeng2 commented:
Bumping this, since it's probably relatively easy to review and merge (b/c it already has been code reviewed once)

bilderbuchi commented on pymeasure/instruments/yokogawa/yokogawags200.py:
```diff
@@ -0,0 +1,147 @@
+                "Otherwise, the Yokogawa will reject the ramp."
+            )
+        if ramp_time < MIN_RAMP_TIME:
+            warnings.warn(
```
Instead of the `warn`, please use a logger. At the top of the file (after imports), add
```
log = logging.getLogger(__name__)
log.addHandler(logging.NullHandler())
```

then here, use `log.warning(...` instead of `warnings.warn(`

bilderbuchi replied:
this reflect usage in other instruments.

bilderbuchi reviewed (changes requested):
Looks good to me, except for one small change. Thank you!

dennisfeng2 commented:
Great, I've refactored to use `log.warning` now! Could you take a look again?

bilderbuchi commented:
Thank you, looks good!
